src/sentry/models/useravatar.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import
+
+import uuid
+
+from PIL import Image
+
+from django.db import models
+
+from sentry.db.models import FlexibleForeignKey, Model
+from sentry.utils.cache import cache
+from sentry.utils.compat import StringIO
+
+
+class UserAvatar(Model):
+    """
+    A UserAvatar associates a User with their avatar photo File
+    and contains their preferences for avatar type.
+    """
+    __core__ = False
+
+    AVATAR_TYPES = (
+        ('letter_avatar', 'LetterAvatar'),
+        ('upload', 'Upload'),
+        ('gravatar', 'Gravatar'),
+    )
+
+    ALLOWED_SIZES = (20, 48, 52, 64, 80, 96, 120)
+
+    user = FlexibleForeignKey('sentry.User', unique=True, related_name='avatar')
+    file = FlexibleForeignKey('sentry.File', unique=True, null=True, on_delete=models.SET_NULL)
+    ident = models.CharField(max_length=36, unique=True, db_index=True, null=True)
+    avatar_type = models.CharField(max_length=20, choices=AVATAR_TYPES, default='letter_avatar')
```

Comment:
On second thought, we could/should use integers here for what we store in the database.

So something like:

```
AVATAR_TYPES = (
    (0, 'LetterAvatar'),
    (1, 'Upload'),
    (2, 'Gravatar'),
)